python/analysis/raster/qgsalignraster.sip

```diff
@@ -1,183 +1,325 @@
+ :rtype: QPointF
+%End
+
+        void dump() const;
+%Docstring
+write contents of the object to standard error stream - for debugging
+%End
+
+        double identify( double mx, double my );
+%Docstring
+Get raster value at the given coordinates (from the first band)
+ :rtype: float
+%End
+
+      protected:
+        int mXSize, mYSize;
+%Docstring
+raster grid size
+%End
 
     };
 
 
-    //! Resampling algorithm to be used (equivalent to GDAL's enum GDALResampleAlg)
-    //! @note RA_Max, RA_Min, RA_Median, RA_Q1 and RA_Q3 are available on GDAL >= 2.0 builds only
     enum ResampleAlg
     {
-      RA_NearestNeighbour, //!< Nearest neighbour (select on one input pixel)
-      RA_Bilinear,       //!< Bilinear (2x2 kernel)
-      RA_Cubic,          //!< Cubic Convolution Approximation (4x4 kernel)
-      RA_CubicSpline,    //!< Cubic B-Spline Approximation (4x4 kernel)
-      RA_Lanczos,        //!< Lanczos windowed sinc interpolation (6x6 kernel)
-      RA_Average,        //!< Average (computes the average of all non-NODATA contributing pixels)
-      RA_Mode,            //!< Mode (selects the value which appears most often of all the sampled points)
-      RA_Max, //!< Maximum (selects the maximum of all non-NODATA contributing pixels)
-      RA_Min, //!< Minimum (selects the minimum of all non-NODATA contributing pixels)
-      RA_Median, //!< Median (selects the median of all non-NODATA contributing pixels)
-      RA_Q1, //!< First quartile (selects the first quartile of all non-NODATA contributing pixels)
-      RA_Q3, //!< Third quartile (selects the third quartile of all non-NODATA contributing pixels)
+      RA_NearestNeighbour,
+      RA_Bilinear,
+      RA_Cubic,
+      RA_CubicSpline,
+      RA_Lanczos,
+      RA_Average,
+      RA_Mode,
+      RA_Max,
+      RA_Min,
+      RA_Median,
+      RA_Q1,
+      RA_Q3,
     };
 
-    //! Definition of one raster layer for alignment
     struct Item
     {
       Item( const QString &input, const QString &output );
 
-      //! filename of the source raster
       QString inputFilename;
-      //! filename of the newly created aligned raster (will be overwritten if exists already)
+%Docstring
+filename of the source raster
+%End
       QString outputFilename;
-      //! resampling method to be used
-      QgsAlignRaster::ResampleAlg resampleMethod;
-      //! rescaling of values according to the change of pixel size
+%Docstring
+filename of the newly created aligned raster (will be overwritten if exists already)
+%End
+      ResampleAlg resampleMethod;
+%Docstring
+resampling method to be used
+%End
       bool rescaleValues;
+%Docstring
+rescaling of values according to the change of pixel size
+%End
 
-      // private part
 
-      //! used for rescaling of values (if necessary)
       double srcCellSizeInDestCRS;
+%Docstring
+used for rescaling of values (if necessary)
+%End
     };
-    typedef QList<QgsAlignRaster::Item> List;
+    typedef QList<Item> List;
```

Comment:
This will need fixing - you'll need to change the header to use `typedef QList<QgsAlignRaster::Item> List;`